library/src/main/java/com/datatorrent/lib/io/fs/HDFSOutputOperator.java

```diff
@@ -0,0 +1,393 @@
+    return currentPartName;
+  }
+
+  /**
+   * Input port for receiving string tuples.
+   */
+  public final transient DefaultInputPort<String> stringInput = new DefaultInputPort<String>()
+  {
+    @Override
+    public void process(String tuple)
+    {
+      processTuple(tuple.getBytes());
+    }
+
+    @Override
+    public StreamCodec<String> getStreamCodec()
+    {
+      if (HDFSOutputOperator.this.stringStreamCodec == null) {
+        return super.getStreamCodec();
+      } else {
+        return stringStreamCodec;
+      }
+    }
+  };
+
+  /**
+   * {@inheritDoc}
+   * 
+   * @return byte[] representation of the given tuple. if input tuple is of type
+   *         byte[] then it is returned as it is. for any other type toString()
+   *         representation is used to generate byte[].
+   */
+  @Override
+  protected byte[] getBytesForTuple(byte[] tuple)
+  {
+    ByteArrayOutputStream bytesOutStream = new ByteArrayOutputStream();
+
+    try {
+      bytesOutStream.write(tuple);
+      bytesOutStream.write(tupleSeparatorBytes);
+      byteCount += bytesOutStream.size();
+      return bytesOutStream.toByteArray();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } finally {
+      try {
+        bytesOutStream.close();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  /**
+   * Initializing per window level fields {@inheritDoc}
+   */
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    byteCount = 0;
+    isNewDataInCurrentWindow = false;
+  }
+
+  /**
+   * {@inheritDoc} Does additional state maintenance for rollover
+   */
+  @Override
+  protected void processTuple(byte[] tuple)
+  {
+    super.processTuple(tuple);
+    isNewDataInCurrentWindow = true;
+
+    if (++currentPartTupleCount == maxTupleCount) {
+      rotateCall(currentPartName);
+    }
+  }
+
+  /**
+   * {@inheritDoc} Does additional checks if file should be rolled over for this
+   * window.
+   */
+  @Override
+  public void endWindow()
+  {
+    super.endWindow();
+
+    if (!isNewDataInCurrentWindow) {
+      ++currentPartIdleWindows;
+    } else {
+      currentPartIdleWindows = 0;
+    }
+
+    if (checkEndWindowFinalization()) {
+      rotateCall(currentPartName);
+    }
+  }
+
+  /**
+   * Rollover check at the endWindow
+   */
+  private boolean checkEndWindowFinalization()
+  {
+    if ((currentPartIdleWindows == maxIdleWindows) && !endOffsets.isEmpty()) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * {@inheritDoc} Handles file rotation along with exception handling
+   * 
+   * @param lastFile
+   */
+  protected void rotateCall(String lastFile)
+  {
+    try {
+      this.rotate(lastFile);
+      currentPartName = String.format(currentPartNameformat, fileName, physicalPartitionId);
+      currentPartIdleWindows = 0;
+      currentPartTupleCount = 0;
+    } catch (IOException ex) {
+      LOG.debug(ex.getMessage());
```

Comment:
why not log the entire exception ? LOG.debug("Exception in file rotation",ex);.
Also, error level log seems more appropriate.